chainer/functions/array/where.py

```diff
@@ -14,9 +14,12 @@ def __init__(self, condition):
         self.condition = condition
 
     def check_type_forward(self, in_types):
-        type_check.expect(in_types.size() == 2)
+        type_check._argname(in_types, ('x', 'y'))
         x_type, y_type = in_types
-        condition = self.condition
+        condition = type_check._make_variable_from_array(
+            # allow scalar `condition`
+            chainer.utils.force_array(self.condition),
```

Comment:
I think it's better to avoid having to call `force_array` explicitly, as it's nonessential and not so cheap.
How about something like `type_check._make_variable_from_array_or_scalar`?